packages/components/src/mobile/link-settings/test/edit.native.js

```diff
@@ -0,0 +1,432 @@
+// noinspection DuplicatedCode
+
+/**
+ * External dependencies
+ */
+import { Clipboard } from 'react-native';
+import { fireEvent, initializeEditor, waitFor } from 'test/helpers';
+/**
+ * WordPress dependencies
+ */
+import { registerCoreBlocks } from '@wordpress/block-library';
+import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Utility function to unregister all core block types previously registered
+ * when staging the Redux Store `beforeAll` integration tests start running.
+ *
+ * This should probably be extracted into a utility module for future tests.
+ */
+const unregisterBlocks = () => {
+	const blocks = getBlockTypes();
+
+	blocks.forEach( ( { name } ) => unregisterBlockType( name ) );
+};
+
+/**
+ * ### TODO
+ * + Try to figure out why I can't `console.log(JSON.stringify(subject.toJSON()))` anymore.
+ */
+describe.each( [
+	[
+		{
+			type: 'core/button',
+			initialHtml: `
+				<!-- wp:button {"style":{"border":{"radius":"5px"}}} -->
+				<div class="wp-block-button">
+					<a class="wp-block-button__link" style="border-radius:5px">Link</a>
+				</div>
+				<!-- /wp:button -->
+			`,
+			toJSON: () => 'core/button',
```

Comment:
@ttahmouch I'm wondering if we need this property as I don't see being exposed and used in the function at line 58, should we remove it?

Reply:
It is actually used at [line 58](https://github.com/ttahmouch/gutenberg/blob/40f5004851908960a835a013086b56f56cb25cb4/packages/components/src/mobile/link-settings/test/edit.native.js#L58) just not in the way you'd think.

> If the value has a toJSON() method, it's responsible to define what data will be serialized.
> https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

`'<LinkSettings/> from %j'` is serialized with the test arguments `toJSON` to make it clear in the console logs that this test suite iteration is for a different block under test, i.e., `core/button` then `core/image`.

I can remove it, but I felt it to be useful. WDYT?